File: FixJson.java
import java.io.*;
import java.nio.file.*;
import java.nio.charset.Charset;

public class FixJson {
    public static void main(String[] args) throws Exception {
        String path = "E:/Java/MinecraftMod/DarkGrey/src/main/resources/assets/dark_grey/data/rpg_items.json";
        String content = new String(Files.readAllBytes(Paths.get(path)), Charset.forName("GBK"));
        content = content.trim();
        if (content.endsWith("]")) {
            content = content.substring(0, content.length() - 1);
        }
        
        String append = ",\n" +
            "    {\n" +
            "      \"id\": \"underground_sun\",\n" +
            "      \"type\": \"道具\",\n" +
            "      \"displayName\": {\n" +
            "        \"zh_CN\": \"地底太阳\",\n" +
            "        \"en_US\": \"Underground Sun\"\n" +
            "      },\n" +
            "      \"texture\": \"dark_grey:underground_sun\",\n" +
            "      \"durability\": 0,\n" +
            "      \"damage\": 90,\n" +
            "      \"components\": [\n" +
            "        {\n" +
            "          \"name\": \"地底太阳\",\n" +
            "          \"params\": {\n" +
            "            \"chargeTicks\": 40,\n" +
            "            \"maxStoredOrbs\": 3,\n" +
            "            \"damageMultiplier\": 5.0,\n" +
            "            \"explosionRadius\": 20.0,\n" +
            "            \"explosionHalfHeight\": 10.0,\n" +
            "            \"projectileSpeed\": 1.8,\n" +
            "            \"projectileLifetime\": 100,\n" +
            "            \"launchCooldownTicks\": 5,\n" +
            "            \"ignoreHurtResistance\": true,\n" +
            "            \"respectWalls\": false,\n" +
            "            \"orbitRadius\": 1.25,\n" +
            "            \"orbitHeight\": -0.2,\n" +
            "            \"orbitSpeed\": 2.0\n" +
            "          }\n" +
            "        }\n" +
            "      ],\n" +
            "      \"enchantments\": \"\"\n" +
            "    }\n" +
            "]";
            
        content += append;
        Files.write(Paths.get(path), content.getBytes(Charset.forName("GBK")));
        System.out.println("JSON fixed perfectly in GBK");
    }
}

File: fix_json.py
import json
import codecs

path = 'src/main/resources/assets/dark_grey/data/rpg_items.json'
with codecs.open(path, 'r', 'utf-8') as f:
    data = json.load(f)

ids = [item['id'] for item in data]
if 'sunflame_round' not in ids:
    data.append({
      'id': 'sunflame_round',
      'type': '材料',
      'displayName': { 'zh_CN': '阳炎弹', 'en_US': 'Sunflame Round' },
      'texture': 'dark_grey:sunflame_round',
      'durability': 0,
      'damage': 0,
      'components': [],
      'maxStackSize': 64,
      'enchantments': ''
    })

if 'slag_eruptor' not in ids:
    data.append({
      'id': 'slag_eruptor',
      'type': '铳',
      'displayName': { 'zh_CN': '熔渣喷发器', 'en_US': 'Slag Eruptor' },
      'texture': 'dark_grey:slag_eruptor',
      'durability': 0,
      'damage': 0,
      'components': [{
          'name': '熔渣喷发器',
          'params': {
            'loadTicksRequired': 40,
            'baseDamage': 50.0,
            'magazineCapacity': 30
          }
      }],
      'maxStackSize': 1,
      'enchantments': ''
    })

with codecs.open(path, 'w', 'utf-8') as f:
    json.dump(data, f, indent=2, ensure_ascii=False)
print('JSON updated via Python.')
